internal/cluster/cluster.go

```diff
@@ -740,3 +741,7 @@ func (m Manager) PermanentClustersDeletion(ctx context.Context, olderThen strfmt
 	}
 	return nil
 }
+func (m Manager) UpdateInstallProgress(ctx context.Context, clusterID strfmt.UUID, progress string, db *gorm.DB) error {
```

Comment:
ack. will remove it.

Reply:
I tried to use the Manager db `m.db` and pass it to the state machine.
As a result, I encountered what seems to be a lock.
The API got blocked with no timeout, when I use `db` as it looks now - it works as expected.

What do you think?

Reply:
transaction is locking the DB to write operations so because you are using transaction in UpdateClusterInstallProgress then you should use the same transaction DB in all the sub operations as well.

Reply:
But i'm not sure that you need a transaction in UpdateClusterInstallProgress, why did you use transaction there?

Reply:
Done.
No longer using db transactions.
Additionally, I'm the `transitionHandler` db